### ISSUE TITLE
PR lists for newsletter using GitHub API

### ISSUE THREAD

**Issue by zmoon** [fortran-lang/fortran-lang.org#258]
Maybe you guys already do something like this, but I wanted to figure out how it could be done, so I wrote a little Python script. Thought I would share, in case it would be helpful for the newsletter.

<details>
<summary>Code and output</summary>

Code:
```python
"""
Summarize PRs for fortran-lang within a certain month.
"""
import datetime
from calendar import monthrange

from github import Github

# Set month
month = 4  # April

# Token (set so that the GitHub API doesn't complain)
token = "???"

# Determine time bounds to check
now = datetime.datetime.now()
_, d_b = monthrange(now.year, month)
t_a = datetime.datetime(now.year, month, 1, 0, 0)
t_b = datetime.datetime(now.year, month, d_b, 23, 59)

g = Github(token, per_page=100)

org = g.get_organization("fortran-lang")


def get_prs(repo="stdlib"):
    repo = org.get_repo(repo)
    
    merged = []  # PRs merged during the month
    for pr in repo.get_pulls("closed", sort="updated", direction="desc"):  # most recently merged first
        if pr.merged:
            if t_a <= pr.merged_at <= t_b:
                merged.append(pr)
            if pr.merged_at < t_a:
                break
            
    wip = []  # WIP PRs (not merged, still open at this time)
    for pr in repo.get_pulls("open"):
        if pr.created_at <= t_b:
            wip.append(pr)
    
    return merged, wip


# %% Print markdown

merged, wip = get_prs()  # takes a while...

print("## Fortran Standard Library\n")
print("Here's what's new in `stdlib`:\n")
for pr in merged:
    print(f"* [#{pr.number}]({pr.html_url}): {pr.title}")

print("\nWork in progress:\n")
for pr in wip:
    print(f"* [#{pr.number}]({pr.html_url}) (WIP): {pr.title}")
```

Output:
```
## Fortran Standard Library

Here's what's new in `stdlib`:

* [#391](https://github.com/fortran-lang/stdlib/pull/391): Add issue templates
* [#367](https://github.com/fortran-lang/stdlib/pull/367): Add Intel compiler workflow for OSX
* [#360](https://github.com/fortran-lang/stdlib/pull/360): Summarize build toolchain workflow and implied rules
* [#388](https://github.com/fortran-lang/stdlib/pull/388): Changed filenames
* [#384](https://github.com/fortran-lang/stdlib/pull/384): Implement starts_with and ends_with functions
* [#336](https://github.com/fortran-lang/stdlib/pull/336): Add functions to convert integer/logical values to character values
* [#343](https://github.com/fortran-lang/stdlib/pull/343): Implement strip and chomp as supplement to trim
* [#349](https://github.com/fortran-lang/stdlib/pull/349): Simplify test makefile

Work in progress:

* [#386](https://github.com/fortran-lang/stdlib/pull/386) (WIP): Start the addition of the module stdlib_sorting
* [#372](https://github.com/fortran-lang/stdlib/pull/372) (WIP): To_Title is corrected
* [#363](https://github.com/fortran-lang/stdlib/pull/363) (WIP): Added sort to stdlib_string_type.f90 file
* [#353](https://github.com/fortran-lang/stdlib/pull/353) (WIP): Initial checkin for a module for tolerant comparison of reals
* [#333](https://github.com/fortran-lang/stdlib/pull/333) (WIP): Provide abstract base class for a string object
* [#313](https://github.com/fortran-lang/stdlib/pull/313) (WIP): legendre polynomials and gaussian quadrature
* [#311](https://github.com/fortran-lang/stdlib/pull/311) (WIP): String list new
* [#286](https://github.com/fortran-lang/stdlib/pull/286) (WIP): Probability Distribution and Statistical Functions -- Beta Distribution Module
* [#278](https://github.com/fortran-lang/stdlib/pull/278) (WIP): Probability Distribution and Statistical Functions -- Gamma Distribution Module
* [#276](https://github.com/fortran-lang/stdlib/pull/276) (WIP): Probability Distribution and Statistical Functions -- Exponential Distribution Module
* [#273](https://github.com/fortran-lang/stdlib/pull/273) (WIP): Probability Distribution and Statistical Functions -- Normal Distribution Module 
* [#272](https://github.com/fortran-lang/stdlib/pull/272) (WIP): Probability Distribution and Statistical Functions -- Uniform Distribution Module
* [#189](https://github.com/fortran-lang/stdlib/pull/189) (WIP): Initial implementation of COO / CSR sparse format
* [#157](https://github.com/fortran-lang/stdlib/pull/157) (WIP): Update CMAKE files
```
</details>

**Comment by milancurcic** [comment]
Thanks @zmoon, I've been writing these by hand, but perhaps others (@jvdp1 @awvwgk) had it scripted. Either way, it be very useful to have a script that works and have it part of the repo. Perhaps it could be even be put into CI as a GH Action that runs on a set day of the month and opens a new PR with a draft newsletter?

**Comment by certik** [comment]
Nice! That's the way to do it.

**Comment by zmoon** [comment]
> Either way, it be very useful to have a script that works and have it part of the repo. Perhaps it could be even be put into CI as a GH Action that runs on a set day of the month and opens a new PR with a draft newsletter?

The latter would certainly be cool. I would be happy to start with the former if there is interest. Where in the repo should such a script go? Also would probably want to have the template as a separate file.

**Comment by zmoon** [comment]
Closed by #261
